mod/profile/views/default/resources/profile/view.php

```diff
@@ -7,7 +7,11 @@
 	$user = elgg_get_logged_in_user_entity();
 }
 
-elgg_entity_gatekeeper($user->guid, 'user');
```

Comment:
Keep the gatekeeper

Reply:
`$user` could be empty (invalid username) and then you get a notice about getting a property on a non object

Reply:
You can throw, but keep the gatekeeper.
I am thinking about GDPR and all the ways we will need to be restricting access to profiles, so keep the gatekeeper in for other purposes.

Reply:
i restored it